CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+is to join one of the channels listed below:
+
+* [Technical Mailing List](http://lists.projectcalico.org/mailman/listinfo/calico-tech_lists.projectcalico.org)
+* [Slack Calico Users Channel](https://calicousers.slack.com) ([Sign up](https://calicousers-slackin.herokuapp.com))
+* IRC - [#calico](https://kiwiirc.com/client/irc.freenode.net/#calico)
+
+## Reporting issues
+
+Before raising an issue with *Calico for containers*, please check for 
+duplicate issues, and read our [Troubleshooting](Troubleshooting.md)
+and our [Frequently Asked Questions](FAQ.md) documents.
+
+If you have a question, please hop on to our IRC or Slack channels.
+
+If you do need to raise an issue, please include any of the following that may
+be relevant to your problem (including too much information is never
+a bad thing):
+
+-  A detailed description of the problem.
+-  The steps to reproduce the problem (e.g. the full list of `calicoctl`
+   commands that were run)
+-  The expected result and actual result.
+-  Versions of appropriate binaries and libraries.  For example, the output from
+   `calicoctl version`, your version of Docker, rkt, Kubernetes etc.
+-  A link to any diagnostics (e.g. if using `calicoctl`, you can gathered
+   diagnostics using `calicoctl diags` - this provides instructions for 
+   uploading the diags bundle to transfer.sh - or alternatively if the 
+   diagnostics contains sensitive information we can set up an alternative
+   method for transfer
+-  If using `calicoctl` the output from `calicoctl status` run on each node
+   might also be useful.
+-  Details of your OS
+-  Environment details such as GCE, bare metal, VirtualBox.
+   
+
+## Contributing: Code and Documentation
+
+For contributing code and documentation we follow the GitHub pull request
+model. 
+
+-  Fork the repository on GitHub
+-  Make changes to your local repository in a separate branch
+-  Test the change
+-  Create a pull request which will then go through a review process by one or
+   more of the core team.
+   
+### Testing your changes
+
+If you create a pull request, our automated UT and STs will be run over your
+change.  We will not accept changes that do not consistently pass our automated
+test suites. It is vital that our master branch be passing tests at all times.
+If you tests are failing the automated tests and you don't believe they should
+be, you may need to rebase your branch off the latest master.
+
+Read the [Building and testing calico-containers images guide](docs/Building.md)
+for details on running the UTs and STs.
+
+Where possible, please add any additional tests to ensure we maintain a healthy
+code and feature coverage.
+
+### Documentation
+
+If your code change requires some documentation updates, please include these
+in the pull request.
+
+In particular changes to the `calicoctl` UX will also require changes to the
+`calcoctl` documentation and possibly some of the sample guides.
+
+### Review and merge
```

Comment:
Actually looking at other sections, the capitalization of sections isn't really consistent.  Nitpicky, but could be nice to have them all capitalized the same.